From d7ea6c5c1387289f89dd83da5c5188327acd1f00 Mon Sep 17 00:00:00 2001
From: Catalin Horghidan <catalin.horghidan@nxp.com>
Date: Thu, 22 Sep 2016 17:30:57 +0300
Subject: [PATCH 217/448] staging: fsl-dpaa2: rtc: update dprtc command ids set for MC v10.x

	Update dprtc command ids in order to work with MC 10.0.0.
	The binary interface compatibility is maintained with MC v9.x.

Signed-off-by: Catalin Horghidan <catalin.horghidan@nxp.com>
---
 drivers/staging/fsl-dpaa2/rtc/dprtc-cmd.h |   62 +++++++++++++++-------------
 1 files changed, 33 insertions(+), 29 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/rtc/dprtc-cmd.h b/drivers/staging/fsl-dpaa2/rtc/dprtc-cmd.h
index 6174975..2655049 100644
--- a/drivers/staging/fsl-dpaa2/rtc/dprtc-cmd.h
+++ b/drivers/staging/fsl-dpaa2/rtc/dprtc-cmd.h
@@ -1,4 +1,4 @@
-/* Copyright 2013-2015 Freescale Semiconductor Inc.
+/* Copyright 2013-2016 Freescale Semiconductor Inc.
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -36,38 +36,42 @@
 #define DPRTC_VER_MAJOR				1
 #define DPRTC_VER_MINOR				0
 
+#define DPRTC_CMD_BASE_VER				0
+#define DPRTC_CMD_ID_OFF				4
+#define DPRTC_CMD_ID(id) (((id) << DPRTC_CMD_ID_OFF) | DPRTC_CMD_BASE_VER)
+
 /* Command IDs */
-#define DPRTC_CMDID_CLOSE				0x800
-#define DPRTC_CMDID_OPEN					0x810
-#define DPRTC_CMDID_CREATE				0x910
-#define DPRTC_CMDID_DESTROY				0x900
-
-#define DPRTC_CMDID_ENABLE				0x002
-#define DPRTC_CMDID_DISABLE				0x003
-#define DPRTC_CMDID_GET_ATTR				0x004
-#define DPRTC_CMDID_RESET				0x005
-#define DPRTC_CMDID_IS_ENABLED				0x006
-
-#define DPRTC_CMDID_SET_IRQ				0x010
-#define DPRTC_CMDID_GET_IRQ				0x011
-#define DPRTC_CMDID_SET_IRQ_ENABLE			0x012
-#define DPRTC_CMDID_GET_IRQ_ENABLE			0x013
-#define DPRTC_CMDID_SET_IRQ_MASK				0x014
-#define DPRTC_CMDID_GET_IRQ_MASK				0x015
-#define DPRTC_CMDID_GET_IRQ_STATUS			0x016
-#define DPRTC_CMDID_CLEAR_IRQ_STATUS			0x017
-
-#define DPRTC_CMDID_SET_CLOCK_OFFSET			0x1d0
-#define DPRTC_CMDID_SET_FREQ_COMPENSATION		0x1d1
-#define DPRTC_CMDID_GET_FREQ_COMPENSATION		0x1d2
-#define DPRTC_CMDID_GET_TIME					0x1d3
-#define DPRTC_CMDID_SET_TIME					0x1d4
-#define DPRTC_CMDID_SET_ALARM					0x1d5
-#define DPRTC_CMDID_SET_PERIODIC_PULSE			0x1d6
-#define DPRTC_CMDID_CLEAR_PERIODIC_PULSE		0x1d7
-#define DPRTC_CMDID_SET_EXT_TRIGGER				0x1d8
-#define DPRTC_CMDID_CLEAR_EXT_TRIGGER			0x1d9
-#define DPRTC_CMDID_GET_EXT_TRIGGER_TIMESTAMP	0x1dA
+#define DPRTC_CMDID_CLOSE                     DPRTC_CMD_ID(0x800)
+#define DPRTC_CMDID_OPEN                      DPRTC_CMD_ID(0x810)
+#define DPRTC_CMDID_CREATE                    DPRTC_CMD_ID(0x910)
+#define DPRTC_CMDID_DESTROY                   DPRTC_CMD_ID(0x900)
+
+#define DPRTC_CMDID_ENABLE                    DPRTC_CMD_ID(0x002)
+#define DPRTC_CMDID_DISABLE                   DPRTC_CMD_ID(0x003)
+#define DPRTC_CMDID_GET_ATTR                  DPRTC_CMD_ID(0x004)
+#define DPRTC_CMDID_RESET                     DPRTC_CMD_ID(0x005)
+#define DPRTC_CMDID_IS_ENABLED                DPRTC_CMD_ID(0x006)
+
+#define DPRTC_CMDID_SET_IRQ                   DPRTC_CMD_ID(0x010)
+#define DPRTC_CMDID_GET_IRQ                   DPRTC_CMD_ID(0x011)
+#define DPRTC_CMDID_SET_IRQ_ENABLE            DPRTC_CMD_ID(0x012)
+#define DPRTC_CMDID_GET_IRQ_ENABLE            DPRTC_CMD_ID(0x013)
+#define DPRTC_CMDID_SET_IRQ_MASK              DPRTC_CMD_ID(0x014)
+#define DPRTC_CMDID_GET_IRQ_MASK              DPRTC_CMD_ID(0x015)
+#define DPRTC_CMDID_GET_IRQ_STATUS            DPRTC_CMD_ID(0x016)
+#define DPRTC_CMDID_CLEAR_IRQ_STATUS          DPRTC_CMD_ID(0x017)
+
+#define DPRTC_CMDID_SET_CLOCK_OFFSET          DPRTC_CMD_ID(0x1d0)
+#define DPRTC_CMDID_SET_FREQ_COMPENSATION     DPRTC_CMD_ID(0x1d1)
+#define DPRTC_CMDID_GET_FREQ_COMPENSATION     DPRTC_CMD_ID(0x1d2)
+#define DPRTC_CMDID_GET_TIME                  DPRTC_CMD_ID(0x1d3)
+#define DPRTC_CMDID_SET_TIME                  DPRTC_CMD_ID(0x1d4)
+#define DPRTC_CMDID_SET_ALARM                 DPRTC_CMD_ID(0x1d5)
+#define DPRTC_CMDID_SET_PERIODIC_PULSE        DPRTC_CMD_ID(0x1d6)
+#define DPRTC_CMDID_CLEAR_PERIODIC_PULSE      DPRTC_CMD_ID(0x1d7)
+#define DPRTC_CMDID_SET_EXT_TRIGGER           DPRTC_CMD_ID(0x1d8)
+#define DPRTC_CMDID_CLEAR_EXT_TRIGGER         DPRTC_CMD_ID(0x1d9)
+#define DPRTC_CMDID_GET_EXT_TRIGGER_TIMESTAMP DPRTC_CMD_ID(0x1dA)
 
 /*                cmd, param, offset, width, type, arg_name */
 #define DPRTC_CMD_OPEN(cmd, dpbp_id) \
-- 
1.7.1

